### DISCUSSION TITLE
Re-enable ESP NOW.

### DISCUSSION THREAD

**Discussion by Pawelinode** [micropython/micropython#14406] [ESP32] [answered]
The first packet is sent successfully. Attempting to send the second one results in an "ESP_ERR_ESPNOW_NOT_FOUND" error. How do I re-enable the ESPNOW interface?
```
import espnow
import network
from time import sleep

e = espnow.ESPNow()
sta = network.WLAN(network.STA_IF) 
sta.active(True)
e.active(True)
e.add_peer(b'\xff' * 6)

print("e:",e.send(b'\xff' * 6, "Hello World!"))
e.active(False)
sleep(0.5)
e.active(True)
sleep(0.5)
print("e:",e.send(b'\xff' * 6, "Hello World!"))
```

**Comment by davefes** [comment]
In an old project I could do this:
```py
while True:
    w0.active(True) #  turn on radio only when needed
 #  if you want to save more battery, set sync=False
 #  at cost of not knowing if message was received.

    w0.config(channel=6) #  to put radio back on ch 6 after lightsleep()

    e0.send(repeater_mac, status, True)

    w0.active(False)

    machine.lightsleep(60000)

```
Maybe something to do with putting WiFi back on-channel??

**Reply by Pawelinode** [reply, marked as answer]
I found the cause of the problem. I did not re-execute e.add_peer(b'\xff' * 6)

**Reply by davefes** [reply]
OK.  I only had `e.add_peer()` once before the while loop.

**Reply by glenn20** [reply]
Yes, `e.active(False);  e.active(True)` will completely reinitialise the espnow software system, including clearing the list of registered peers.

There is not usually anything to be gained by de-initialising the espnow stack between sending messages.
